nexus/src/app/background/sync_switch_port_settings.rs

```diff
@@ -0,0 +1,1236 @@
+                            None => {
+                                error!(log, "sled-agent client is missing, cannot send updates"; "location" => %location);
+                                continue;
+                            },
+                        };
+
+                    info!(
+                        &log,
+                        "applying SMF config uplink updates to switch zone";
+                        "switch_location" => ?location,
+                        "config" => ?config,
+                    );
+                    if let Err(e) = client
+                        .uplink_ensure(&sled_agent_client::types::SwitchPorts {
+                            uplinks: config.clone(),
+                        })
+                        .await
+                    {
+                        error!(
+                            log,
+                            "error while applying smf updates to switch zone";
+                            "location" => %location,
+                            "error" => %e,
+                        );
+                    }
+                }
+
+                //
+                // calculate and apply BGP changes
+                //
+
+                // build a list of desired settings for each switch
+                let mut desired_bgp_configs: HashMap<
+                    SwitchLocation,
+                    Vec<ApplyRequest>,
+                > = HashMap::new();
+
+                // we currently only support one bgp config per switch
+                let mut switch_bgp_config: HashMap<SwitchLocation, (Uuid, BgpConfig)> = HashMap::new();
+
+                // Prefixes are associated to BgpConfig via the config id
+                let mut bgp_announce_prefixes: HashMap<Uuid, Vec<Prefix4>> = HashMap::new();
+
+                let mut bootstore_bgp_peer_info: Vec<(SwitchPortBgpPeerConfig, u32, Ipv4Addr)> = vec![];
+
+                for (location, port, change) in &changes {
+                    let PortSettingsChange::Apply(settings) = change else {
+                        continue;
+                    };
+
+                    // desired peer configurations for a given switch port
+                    let mut peers: HashMap<String, Vec<BgpPeerConfig>> = HashMap::new();
+
+                    for peer in &settings.bgp_peers {
+                        let bgp_config_id = peer.bgp_config_id;
+
+                        // since we only have one bgp config per switch, we only need to fetch it once
+                        let bgp_config = match switch_bgp_config.entry(*location) {
+                            Entry::Occupied(occupied_entry) => {
+                                let (existing_id, existing_config) = occupied_entry.get().clone();
+                                // verify peers don't have differing configs
+                                if existing_id != bgp_config_id {
+                                    // should we flag the switch and not do *any* updates to it?
```

Comment:
I think logging and error and continuing is good here. We should think about how to get this information to operators. It should be this case that the RPW will converge to the intended state when remedial action is taken by an operator.